### DISCUSSION TITLE
Rendering Video Texture

### DISCUSSION THREAD

**Discussion by davidAnPrintell** [sceneview/sceneform-android#348] [Q&A] [answered]
Hey!
I face some kind of issue while experimenting with image targets for my company.

I'm not exactly sure if thats an issue or just a misunderstanding by my side, so im creating this discussion.

#### Issue description

The video texture renders too bright while shading is set to unlit. Neither the material.postLightingColor attribute nor material.emissive has any effect on this.
If I increase the postlightingColor the videotexture only gets brighter( of course).
I guess this has to do with the "general colorgrading"?

#### Steps to reproduce the issue

1.  Start sample image tracking project
2. Change "matrix video"
3. Track Image

#### What's the expected result?

<div style="display:flex">
<img src="https://user-images.githubusercontent.com/96430005/160291347-33557894-cdc2-47b5-b6c2-8e54fb4b16eb.PNG" width="300">
<img src="https://user-images.githubusercontent.com/96430005/160291353-5f374094-529c-4141-8f66-3a68485eb952.PNG" width="300">
</div>

#### What's the actual result?

<div style="display:flex">
<img src="https://user-images.githubusercontent.com/96430005/160291209-ff3e5d88-c9aa-42b7-9055-ebc14aa018be.jpg" width="300">
<img src="https://user-images.githubusercontent.com/96430005/160291321-b74b1af1-eee0-43e4-8102-64c7e7b19997.jpg" width="300">
</div>

#### Additional code details
Material :
```kotlin
 MaterialBuilder materialBuilder = new MaterialBuilder()
                .platform(MaterialBuilder.Platform.MOBILE)
                .name("External Video Material")
                .require(MaterialBuilder.VertexAttribute.UV0)
                .shading(MaterialBuilder.Shading.UNLIT)
                .doubleSided(true)
                .samplerParameter(MaterialBuilder.SamplerType.SAMPLER_EXTERNAL, MaterialBuilder.SamplerFormat.FLOAT, MaterialBuilder.ParameterPrecision.DEFAULT, "videoTexture")
                .optimization(MaterialBuilder.Optimization.NONE);

        MaterialPackage plainVideoMaterialPackage = materialBuilder
                .blending(MaterialBuilder.BlendingMode.OPAQUE)
                .material("void material(inout MaterialInputs material) {\n" +
                        "    prepareMaterial(material);\n" +
                        "    material.baseColor = texture(materialParams_videoTexture, getUV0()).rgba;\n" +
                        "    material.postLightingColor = float4(0.0);\n" +
                        "    material.emissive = float4(0.0, 0.0, 0.0, 1.0);\n" +
                        "}\n")
                .build(filamentEngine);
```

Configs
```kotlin
  config.updateMode = Config.UpdateMode.LATEST_CAMERA_IMAGE
                    config.focusMode = Config.FocusMode.AUTO
                    config.planeFindingMode = Config.PlaneFindingMode.DISABLED
                    config.augmentedImageDatabase = imageDatabase
                    config.lightEstimationMode = Config.LightEstimationMode.DISABLED
                    session.configure(config)
                    arSceneView?.lightEstimationConfig = LightEstimationConfig.DISABLED
                    arSceneView?.renderer!!.filamentView.colorGrading = ColorGrading.Builder().toneMapping(ColorGrading.ToneMapping.FILMIC)
                        .build(EngineInstance.getEngine().filamentEngine)
                    arSceneView?.setSession(session)
```

**Comment by grassydragon** [comment, marked as answer]
Hi!
If you think that it is related to tone mapping, you can cancel the effect of tone mapping using the `inverseTonemap` function: https://google.github.io/filament/Materials.html#materialdefinitions/shaderpublicapis/fragmentonly

``` glsl
material.baseColor.rgb = inverseTonemap(material.baseColor.rgb)
```

**Reply by davidAnPrintell** [reply]
Thanks for your reply. It fixed the problem immediately
